### PR TITLE
[ML] Restrict JobServiceType to jupyter_lab, ssh, tensor_board, vs_code and perform required conversion before sending to rest API

### PR DESCRIPTION
**Pull request by TonyJ1**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/job_service.py [hunk 1]
```diff
@@ -37,7 +38,9 @@ def __init__(
         self,
         *,
         endpoint: Optional[str] = None,
-        job_service_type: Optional[Literal["JupyterLab", "SSH", "TensorBoard", "VSCode"]] = None,
+        # TODO: Is VS Code allowed here?
+        # TODO: As this is a breaking change, should we still allow the old format (any-case) and deprecate in the next release?
+        job_service_type: Optional[Literal["jupyter_lab", "ssh", "tensor_board", "vs-code"]] = None,
```
Comment (srsaggam) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/job_service.py:43]: any reason this is vs-code and not vs_code?

Comment (srsaggam) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/job_service.py:43]: This feature was in private preview only. So breaking change should be okay I guess at this point if we want to stick to consistency?

File: sdk/ml/azure-ai-ml/tests/test_configs/pipeline_jobs/helloworld_pipeline_job_with_node_services_inline_job.yml [hunk 2]
```diff
@@ -13,11 +13,11 @@ jobs:
     command: echo Hello World & sleep 1h
     environment: azureml:AzureML-sklearn-0.24-ubuntu18.04-py37-cpu:1
     services:
-      "my_jupyter":
-        job_service_type: "Jupyter" # Jupyter Notebook
+      "my_ssh":
+        job_service_type: "Jupyter"
```
Comment (srsaggam) [inline sdk/ml/azure-ai-ml/tests/test_configs/pipeline_jobs/helloworld_pipeline_job_with_node_services_inline_job.yml:17]: We should remove Jupyter type as it is not supported.
Would pipeline team needs any changes following this?

File: sdk/ml/azure-ai-ml/tests/test_configs/pipeline_jobs/helloworld_pipeline_job_with_node_services_inline_job.yml [hunk 3]
```diff
@@ -13,11 +13,11 @@ jobs:
     command: echo Hello World & sleep 1h
     environment: azureml:AzureML-sklearn-0.24-ubuntu18.04-py37-cpu:1
     services:
-      "my_jupyter":
-        job_service_type: "Jupyter" # Jupyter Notebook
+      "my_ssh":
```
Comment (srsaggam) [inline sdk/ml/azure-ai-ml/tests/test_configs/pipeline_jobs/helloworld_pipeline_job_with_node_services_inline_job.yml:16]: the type is jupyter and name is "my_ssh"

**Review by srsaggam — APPROVED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/job_service.py [hunk 4]
```diff
@@ -37,7 +38,7 @@ def __init__(
         self,
         *,
         endpoint: Optional[str] = None,
-        job_service_type: Optional[Literal["JupyterLab", "SSH", "TensorBoard", "VSCode"]] = None,
+        job_service_type: Optional[Literal["jupyter_lab", "ssh", "tensor_board", "vs_code"]] = None,
```
Comment (luigiw) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/job_service.py:41]: Should this just be called `type`?

**Review by luigiw — APPROVED** [review]

**Pull request merged**